test/core/compression/message_compress_fuzzer.cc

```diff
@@ -0,0 +1,36 @@
+#include <fuzzer/FuzzedDataProvider.h>
+#include <grpc/grpc.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "src/core/lib/compression/message_compress.h"
+#include "src/core/lib/security/credentials/credentials.h"
+#include "test/core/util/memory_counters.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  FuzzedDataProvider data_provider(data, size);
+  const auto compression_algorithm =
+      static_cast<grpc_message_compression_algorithm>(
+          data_provider.ConsumeIntegralInRange<int>(
```

Comment:
> what's the value in using the fuzzed input to choose the compression algorithm as opposed to testing each compression algorithm

Mostly using less code (no need for extra build targets/files) and it doesn't come at a very big readability compromise. Also, if a new compression algorithm is ever added, it gets fuzzing for free.

> just using a random number generator bounded between 0 and 2?

If a fuzzer finds a bug, it should be able to deterministically reproduce it with a given input (using an RNG will make it a flaky failure); in a similar vein, because the fuzzing that this will set up is coverage-guided, the fuzzing infrastructure will be able to explore all the compression options with the knowledge that it's doing so.